lib/jxl/encode.cc

```diff
@@ -1588,6 +1634,53 @@ JxlEncoderStatus JxlEncoderAddJPEGFrame(
   return JXL_ENC_SUCCESS;
 }
 
+static bool CanDoFastLossless(const JxlEncoderFrameSettings* frame_settings,
+                              const JxlPixelFormat* pixel_format,
+                              bool has_alpha) {
+  if (!frame_settings->values.lossless) {
+    return false;
+  }
+  if (frame_settings->values.frame_index_box) {
+    return false;
+  }
+  if (frame_settings->values.cparams.speed_tier != jxl::SpeedTier::kLightning) {
+    return false;
+  }
+  if (frame_settings->values.image_bit_depth.type ==
+          JxlBitDepthType::JXL_BIT_DEPTH_CUSTOM &&
+      frame_settings->values.image_bit_depth.bits_per_sample !=
+          frame_settings->enc->metadata.m.bit_depth.bits_per_sample) {
+    return false;
+  }
+  if (!frame_settings->values.frame_name.empty()) {
+    return false;
```

Comment:
I don't understand why the frame name needs to be empty. At least add a TODO here, because I think the main use case for fast lossless encoding is in authoring workflows (gimp, krita, photoshop etc) where you may very well have named layers. Maybe just make it work already :)